images/gha-runner-scale-set-controller/tests/main.tf

```diff
@@ -0,0 +1,70 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" {
+  input = var.digest
+}
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_k3s" "this" {
+  name      = "gha-runner-scale-set-controller"
+  inventory = data.imagetest_inventory.this
+
+  sandbox = {
+    mounts = [
+      {
+        source      = path.module
+        destination = "/tests"
+      }
+    ]
+  }
+}
+
+module "helm" {
+  source    = "../../../tflib/imagetest/helm"
+  chart     = "oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller"
+  name      = "gha-runner-scale-set-controller"
+  namespace = "arc-systems"
+  values = {
+
+    image = {
+      repository = data.oci_string.ref.registry_repo
+      tag        = data.oci_string.ref.pseudo_tag
+    }
+
+  }
+}
+
+resource "imagetest_feature" "basic" {
```

Comment:
We'll need to expand the testing - anything else we can do here to automate? If not what manual tests can we do and can we create a TESTING.md with the steps?

Reply:
I built the image you may use it for testing, either do I: `ghcr.io/developer-guy/gha-runner-scale-set-controller:latest`

Reply:
Getting a silent failure on runner deployments.  The manager works, the runners do not deploy

- values.yaml for the manager
- [pod logs](#podlogs)
- values.yaml for the runners

### values.yaml

```yaml
# Default values for gha-runner-scale-set-controller.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.
labels: {}

# leaderElection will be enabled when replicaCount>1,
# So, only one replica will in charge of reconciliation at a given time
# leaderElectionId will be set to {{ define gha-runner-scale-set-controller.fullname }}.
replicaCount: 1

image:
  repository: ghcr.io
  name: developer-guy/gha-runner-scale-set-controller
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: latest

imagePullSecrets: []
nameOverride: ""
fullnameOverride: ""

env:
## Define environment variables for the controller pod
#  - name: "ENV_VAR_NAME_1"
#    value: "ENV_VAR_VALUE_1"
#  - name: "ENV_VAR_NAME_2"
#    valueFrom:
#      secretKeyRef:
#        key: ENV_VAR_NAME_2
#        name: secret-name
#        optional: true

serviceAccount:
  # Specifies whether a service account should be created for running the controller pod
  create: true
  # Annotations to add to the service account
  annotations: {}
  # The name of the service account to use.
  # If not set and create is true, a name is generated using the fullname template
  # You can not use the default service account for this.
  name: ""

podAnnotations: {}

podLabels: {}

podSecurityContext: {}
# fsGroup: 2000

securityContext: {}
# capabilities:
#   drop:
#   - ALL
# readOnlyRootFilesystem: true
# runAsNonRoot: true
# runAsUser: 1000

resources: {}
## We usually recommend not to specify default resources and to leave this as a conscious
## choice for the user. This also increases chances charts run on environments with little
## resources, such as Minikube. If you do want to specify resources, uncomment the following
## lines, adjust them as necessary, and remove the curly braces after 'resources:'.
# limits:
#   cpu: 100m
#   memory: 128Mi
# requests:
#   cpu: 100m
#   memory: 128Mi

nodeSelector: {}

tolerations: []

affinity: {}

# Mount volumes in the container.
volumes: []
volumeMounts: []

# Leverage a PriorityClass to ensure your pods survive resource shortages
# ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
# PriorityClass: system-cluster-critical
priorityClassName: ""

## If `metrics:` object is not provided, or commented out, the following flags 
## will be applied the controller-manager and listener pods with empty values: 
## `--metrics-addr`, `--listener-metrics-addr`, `--listener-metrics-endpoint`. 
## This will disable metrics.
##
## To enable metrics, uncomment the following lines.
# metrics:
#   controllerManagerAddr: ":8080"
#   listenerAddr: ":8080"
#   listenerEndpoint: "/metrics"

flags:
  ## Log level can be set here with one of the following values: "debug", "info", "warn", "error".
  ## Defaults to "debug".
  logLevel: "debug"
  ## Log format can be set with one of the following values: "text", "json"
  ## Defaults to "text"
  logFormat: "text"

  ## Restricts the controller to only watch resources in the desired namespace.
  ## Defaults to watch all namespaces when unset.
  # watchSingleNamespace: ""

  ## Defines how the controller should handle upgrades while having running jobs.
  ##
  ## The strategies available are:
  ## - "immediate": (default) The controller will immediately apply the change causing the
  ##   recreation of the listener and ephemeral runner set. This can lead to an
  ##   overprovisioning of runners, if there are pending / running jobs. This should not
  ##   be a problem at a small scale, but it could lead to a significant increase of
  ##   resources if you have a lot of jobs running concurrently.
  ##
  ## - "eventual": The controller will remove the listener and ephemeral runner set
  ##   immediately, but will not recreate them (to apply changes) until all
  ##   pending / running jobs have completed.
  ##   This can lead to a longer time to apply the change but it will ensure
  ##   that you don't have any overprovisioning of runners.
  updateStrategy: "immediate"
```

### podlogs

```
2024-04-16T17:57:19Z	INFO	Update strategy set to:	{"updateStrategy": "immediate"}
2024-04-16T17:57:19Z	INFO	AutoscalingListener logging parameters changed	{"LogLevel": "debug", "LogFormat": "text"}
2024-04-16T17:57:19Z	INFO	Registering scale set metrics
2024-04-16T17:57:19Z	INFO	starting manager
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.AutoscalingRunnerSet"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunnerSet"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.AutoscalingListener"}
2024-04-16T17:57:19Z	INFO	Starting Controller	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunnerSet"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunner"}
2024-04-16T17:57:19Z	INFO	Starting Controller	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "source": "kind source: *v1alpha1.EphemeralRunner"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "source": "kind source: *v1.Pod"}
2024-04-16T17:57:19Z	INFO	Starting Controller	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1alpha1.AutoscalingListener"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.Pod"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.ServiceAccount"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.Role"}
2024-04-16T17:57:19Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.RoleBinding"}
2024-04-16T17:57:19Z	INFO	Starting Controller	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener"}
2024-04-16T17:57:19Z	INFO	Starting workers	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "worker count": 1}
2024-04-16T17:57:19Z	INFO	Starting workers	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "worker count": 1}
2024-04-16T17:57:19Z	INFO	Starting workers	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "worker count": 1}
2024-04-16T17:57:19Z	INFO	Starting workers	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "worker count": 1}
```

Runner set values.yaml (`local-defaults.yml` is the file name)

```yaml
## githubConfigUrl is the GitHub url for where you want to configure runners
## ex: https://github.com/myorg/myrepo or https://github.com/myorg

# These runners test and attach themselves to this repo
# ** change this to what you really want **
githubConfigUrl: "https://github.com/some-natalie/kubernoodles"

## githubConfigSecret is the k8s secrets to use when auth with GitHub API.
## You can choose to use GitHub App or a PAT token
githubConfigSecret: 
  ## GitHub Apps Configuration
  # NOTE: IDs MUST be strings, use quotes
  github_app_id: ""
  github_app_installation_id: ""
  github_app_private_key: |
    -----BEGIN RSA PRIVATE KEY-----
    -----END RSA PRIVATE KEY-----

  ### GitHub PAT Configuration
  # github_token: ""

## maxRunners is the max number of runners the auto scaling runner set will scale up to.
maxRunners: 5

## minRunners is the min number of runners the auto scaling runner set will scale down to.
minRunners: 1

# runnerGroup: "default"

template:
  spec:
    containers:
      - name: runner
        image: ghcr.io/actions/actions-runner:latest
        command: ["/home/runner/run.sh"]
```

Reply:
# Upstream testing

This doesn't appear on the upstream project.  Using the same `local-defaults.yml` for the runner deployment as above.

```console
ᐅ helm install arc --namespace "arc-systems" --create-namespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller
Pulled: ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller:0.9.0
Digest: sha256:06b038d924d82eb1f803fd7dbde7a6674e4aea83f681815cb3cffb610e424cff
NAME: arc
LAST DEPLOYED: Tue Apr 16 12:23:16 2024
NAMESPACE: arc-systems
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing gha-runner-scale-set-controller.

Your release is named arc.

WARNING: Older version of the listener (githubrunnerscalesetlistener) is deprecated and will be removed in the future gha-runner-scale-set-0.10.0 release. If you are using environment variable override to force the old listener, please remove the environment variable and use the new listener (ghalistener) instead.

ᐅ k create namespace ghec-runners
namespace/ghec-runners created

ᐅ helm install defaults --namespace "ghec-runners" -f local-defaults.yml oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set --version 0.9.0
Pulled: ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set:0.9.0
Digest: sha256:f76dbf77a28206f030ab3165360cc6c028b13bcefde26284e6959323176e803e
NAME: defaults
LAST DEPLOYED: Tue Apr 16 12:23:35 2024
NAMESPACE: ghec-runners
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing gha-runner-scale-set.

Your release is named defaults.
```

<img width="1308" alt="Screenshot 2024-04-16 at 12 25 19 PM" src="https://github.com/chainguard-images/images/assets/9995618/bae525a8-2ea7-4705-b65b-06a58eb8ecdc">

<https://github.com/some-natalie/kubernoodles/actions/runs/8710660987> is it scaling out to 25 runners on a workflow.

Logs below :)

[manager.txt](https://github.com/chainguard-images/images/files/15001239/manager.txt)
[scale-set-listener.txt](https://github.com/chainguard-images/images/files/15001240/scale-set-listener.txt)
[runner.txt](https://github.com/chainguard-images/images/files/15001241/runner.txt)

Reply:
Same error on the latest rebuild, but it says the deployment of runners succeeded.

```
ᐅ helm install defaults --namespace "ghec-runners" -f local-defaults.yml oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set
Pulled: ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set:0.9.0
Digest: sha256:f76dbf77a28206f030ab3165360cc6c028b13bcefde26284e6959323176e803e
NAME: defaults
LAST DEPLOYED: Tue Apr 16 12:49:12 2024
NAMESPACE: ghec-runners
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing gha-runner-scale-set.

Your release is named defaults.
```

But here's the manager's logs:

```
2024-04-16T18:48:10Z	INFO	Update strategy set to:	{"updateStrategy": "immediate"}
2024-04-16T18:48:10Z	INFO	AutoscalingListener logging parameters changed	{"LogLevel": "debug", "LogFormat": "text"}
2024-04-16T18:48:10Z	INFO	Registering scale set metrics
2024-04-16T18:48:10Z	INFO	starting manager
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "source": "kind source: *v1alpha1.EphemeralRunner"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "source": "kind source: *v1.Pod"}
2024-04-16T18:48:10Z	INFO	Starting Controller	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunnerSet"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunner"}
2024-04-16T18:48:10Z	INFO	Starting Controller	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.AutoscalingRunnerSet"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.EphemeralRunnerSet"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "source": "kind source: *v1alpha1.AutoscalingListener"}
2024-04-16T18:48:10Z	INFO	Starting Controller	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1alpha1.AutoscalingListener"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.Pod"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.ServiceAccount"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.Role"}
2024-04-16T18:48:10Z	INFO	Starting EventSource	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "source": "kind source: *v1.RoleBinding"}
2024-04-16T18:48:10Z	INFO	Starting Controller	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener"}
2024-04-16T18:48:11Z	INFO	Starting workers	{"controller": "ephemeralrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunnerSet", "worker count": 1}
2024-04-16T18:48:11Z	INFO	Starting workers	{"controller": "ephemeral-runner-controller", "controllerGroup": "actions.github.com", "controllerKind": "EphemeralRunner", "worker count": 1}
2024-04-16T18:48:11Z	INFO	Starting workers	{"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "worker count": 1}
2024-04-16T18:48:11Z	INFO	Starting workers	{"controller": "autoscalinglistener", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingListener", "worker count": 1}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Deleting resources	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Finalizing ephemeral runner	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up the runner pod	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Pod is deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up the runner jitconfig secret	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Actual Ephemeral runner counts	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}, "count": 1}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Clean up runner counts	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}, "pending": 0, "running": 0, "finished": 0, "failed": 0, "deleting": 1}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Cleanup finished or failed ephemeral runners	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Waiting for resources to be deleted	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting all ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "count": 1}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Skip ephemeral runner set since it is already marked for deletion	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-9nc76"}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Waiting for ephemeral runner sets to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting all ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "count": 1}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Skip ephemeral runner set since it is already marked for deletion	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-9nc76"}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Waiting for ephemeral runner sets to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Deleting the jitconfig secret	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Waiting for ephemeral runner owned resources to be deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Finalizing ephemeral runner	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up the runner pod	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Pod is deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up the runner jitconfig secret	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Secret is deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up runner linked pods	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Runner-linked pods are deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Cleaning up runner linked secrets	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Runner-linked secrets are deleted	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Removing finalizer	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunner	Successfully removed finalizer after cleanup	{"ephemeralrunner": {"name":"defaults-9nc76-runner-99nqc","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Deleting resources	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Actual Ephemeral runner counts	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}, "count": 0}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	All ephemeral runners are deleted	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Removing finalizer	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	EphemeralRunnerSet	Successfully removed finalizer after cleanup	{"ephemeralrunnerset": {"name":"defaults-9nc76","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	All ephemeral runner sets are deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	AutoscalingRunnerSet	Deleting the runner scale set from Actions service	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:11Z	INFO	actions-clients	retrieve actions client	{"githubConfigURL": "https://github.com/some-natalie/kubernoodles", "namespace": "ghec-runners"}
2024-04-16T18:48:11Z	INFO	actions-clients	creating new client	{"githubConfigURL": "https://github.com/some-natalie/kubernoodles", "namespace": "ghec-runners"}
2024-04-16T18:48:11Z	INFO	actions-clients	successfully created new client	{"githubConfigURL": "https://github.com/some-natalie/kubernoodles", "namespace": "ghec-runners"}
2024-04-16T18:48:11Z	INFO	actions-clients	refreshing token	{"githubConfigUrl": "https://github.com/some-natalie/kubernoodles"}
2024-04-16T18:48:11Z	INFO	actions-clients	getting access token for GitHub App auth	{"accessTokenURL": "https://api.github.com/app/installations/33572463/access_tokens"}
2024-04-16T18:48:11Z	INFO	actions-clients	getting runner registration token	{"registrationTokenURL": "https://api.github.com/repos/some-natalie/kubernoodles/actions/runners/registration-token"}
2024-04-16T18:48:11Z	INFO	actions-clients	getting Actions tenant URL and JWT	{"registrationURL": "https://api.github.com/actions/runner-registration"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Deleted the runner scale set from Actions service	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-binding-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-service-account-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removed finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Waiting for dependent resources to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	All ephemeral runner sets are deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-binding-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-service-account-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	No permission service account has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from GitHub secret	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removed finalizer from GitHub secret	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Waiting for dependent resources to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	All ephemeral runner sets are deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-binding-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-service-account-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	No permission service account has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from GitHub secret	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	GitHub secret has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from manager role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removed finalizer from manager role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Waiting for dependent resources to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	All ephemeral runner sets are deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-binding-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-service-account-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	No permission service account has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from GitHub secret	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	GitHub secret has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from manager role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Manager role binding has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from manager role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removed finalizer from manager role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Waiting for dependent resources to be deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Deleting resources	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up the listener	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Listener is deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Cleaning up ephemeral runner sets	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	All ephemeral runner sets are deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-binding-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-role-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Skipping cleaning up kubernetes mode role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "reason": "annotation key \"actions.github.com/cleanup-kubernetes-mode-service-account-name\" not present"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from no permission service account	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	No permission service account has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-no-permission"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from GitHub secret	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	GitHub secret has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-github-secret"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from manager role binding	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Manager role binding has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer from manager role	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Manager role has already been deleted	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "name": "defaults-gha-rs-manager"}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Removing finalizer	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:12Z	INFO	AutoscalingRunnerSet	Successfully removed finalizer after cleanup	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}}
2024-04-16T18:48:34Z	INFO	AutoscalingRunnerSet	Autoscaling runner set version doesn't match the build version. Deleting the resource.	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "targetVersion": "NA", "actualVersion": "0.9.0"}
2024-04-16T18:49:12Z	INFO	AutoscalingRunnerSet	Autoscaling runner set version doesn't match the build version. Deleting the resource.	{"autoscalingrunnerset": {"name":"defaults","namespace":"ghec-runners"}, "targetVersion": "NA", "actualVersion": "0.9.0"}
```